include/seqan3/alphabet/structure/dot_bracket3.hpp

```diff
@@ -0,0 +1,324 @@
+    {
+        return _value >= rhs._value;
+    }
+    //!\}
+
+protected:
+    //!\privatesection
+    /*!\brief The internal type is a strictly typed enum.
+     *
+     * This is done to prevent aggregate initialization from numbers and/or chars.
+     * It is has the drawback that it also introduces a scope which in turn makes
+     * the static "letter values " members necessary.
+     */
+    enum struct internal_type : rank_type
+    {
+        UNPAIRED,
+        BRACKETL,
+        BRACKETR,
+        UNKNOWN = UNPAIRED
+    };
+
+    //!\brief Value to char conversion table.
+    static constexpr char_type value_to_char[value_size]
+    {
+        '.',
+        '(',
+        ')'
+    };
+
+    //!\brief Char to value conversion table.
+    static constexpr std::array<internal_type, 256> char_to_value
+    {
+        [] () constexpr
+        {
+            using in_t = internal_type;
+            std::array<in_t, 256> ret{};
+
+            // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
+            for (auto & c : ret)
+                c = in_t::UNKNOWN;
+
+            // canonical
+            ret['.'] = in_t::UNPAIRED;
+            ret['('] = in_t::BRACKETL;
+            ret[')'] = in_t::BRACKETR;
+
+            // iupac characters are implicitly "UNKNOWN"
+            return ret;
+        } ()
+    };
+
+public:
+    //!\privatesection
+    //!\brief The data member.
+    internal_type _value;
+    //!\publicsection
+};
+
+constexpr db3 db3::NP{internal_type::UNPAIRED};
+constexpr db3 db3::BL{internal_type::BRACKETL};
+constexpr db3 db3::BR{internal_type::BRACKETR};
+constexpr db3 db3::NA{internal_type::UNKNOWN};
+
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+
+//!\brief seqan3::db3 is defined as being a structure alphabet.
+//!\ingroup structure
+template <>
+struct is_structure<db3> : public std::true_type
+{};
+
+} // namespace seqan3::detail
+
+#ifndef NDEBUG
+static_assert(seqan3::alphabet_concept<seqan3::db3>);
+static_assert(seqan3::structure_concept<seqan3::db3>);
+#endif
+
+// ------------------------------------------------------------------
+// containers
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+//!\brief Alias for an std::vector of seqan3::db3.
+//!\relates db3
+using db3_vector = std::vector<db3>;
+
+
+/*!\brief Alias for an std::basic_string of seqan3::db3.
+ * \relates db3
+ *
+ * \attention
+ * Note that we recommend using seqan3::db3_vector instead of db3_string in almost all situations.
+ * While the C++ style operations on the string are well supported, you should not access the internal c-string
+ * and should not use C-Style operations on it, e.g. the `char_traits::strlen` function will not return the
+ * correct length of the string (while the `.size()` returns the correct value).
+ */
+using db3_string = std::basic_string<db3, std::char_traits<db3>>;
+
+} // namespace seqan3
```

Comment:
I would remove these shortcuts... Let's only have them for the most used data types like dna and rna..